test/test.js

```diff
@@ -0,0 +1,212 @@
+/*!
+ * Copyright (c) 2018 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const axios = require('axios');
+const chai = require('chai');
+const config = require('./config');
+const fs = require('fs');
+const helpers = require('./helpers');
+const https = require('https');
+const mockData = require('./mock-data');
+const mockServer = require('./mock-server');
+const should = chai.should();
+
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
+
+let server = null;
+
+describe('http-signature-middleware', () => {
+  before(() => {
+    const key = fs.readFileSync('./test/basic-server.key');
+    const cert = fs.readFileSync('./test/basic-server.crt');
+    server = https.createServer({cert, key}, mockServer.createServer());
+    server.listen(config.port);
```

Comment:
Should this (and the after/close below?) be using callbacks? Something like:
```js
before(callback => {
  ...
  server.listen(config.port, callback);
})
```
Perhaps same for the after/close call below too.

Reply:
done.